x-pack/plugins/integration_assistant/public/components/create_integration/create_integration_assistant/steps/data_stream_step/sample_logs_input.tsx

```diff
@@ -127,20 +159,35 @@ const parseLogsContent = (fileContent: string): ParseLogsResult => {
         if (parseMultilineNDJSONError instanceof RangeError) {
           return { error: i18n.LOGS_SAMPLE_ERROR.TOO_LARGE_TO_PARSE };
         }
+        // This is an unknown format, so split into lines and return no samplesFormat.
+        const fileLines = fileContent.split('\n').filter((line) => line.trim() !== '');
+        if (fileLines.length === 0) {
+          return { error: i18n.LOGS_SAMPLE_ERROR.EMPTY };
+        }
+
+        const logSamples = _truncateLogSample(fileLines);
```

Comment:
This function name is a bit misleading, it is not only truncating, it always shuffles the array, and only sometimes truncates. Could we use a different name like `prepareParsedLogs` or something?

Also, we could potentially skip everything if the size is < `MaxLogsSampleRows`, right?

Reply:
Good point; I'm quite bad with TS naming. How about `select` as a verb? (I'd love `sample`, but we already use it as a noun). I made some other renames as well.

> we could potentially skip everything if the size is < MaxLogsSampleRows, right?
No, we still shuffle in that case as well. As the function doc says:

 * In all cases it will also:
 *  - shuffle the parsed logs sample using the reproducible shuffle algorithm;
 *  - return no more than MaxLogsSampleRows entries.

Reply:
Okay, yes I agree "_sample_" would be perfect, but it would confuse people, "_select_" is also a good name. 

I also realized that the current function implementation is mutating the `fileLines` array parameter, which is the only thing I don't quite like. Have you considered using the lodash [shuffle](https://lodash.com/docs/4.17.15#shuffle) function? It returns a new array without mutating the parameter, or the [sampleSize](https://lodash.com/docs/4.17.15#sampleSize), which seems to do everything we need.

Reply:
I considered `lodash` functions originally, but they call `Math.random` and are thus not reproducible unless one sets a global seed – which would be an overkill.

I think I'll just call `array.length =  ` in the generic sample function and document that it modifies the original array instead of creating a new one. The only thing I need to make sure is that the memory is released. E.g. if we had 1_000_000 lines in `fileLines` and reduced them to 10 we want to be sure that the other 999_990 lines are released.

Reply:
Why do we need to make it predictable? 

> I think I'll just call array.length =   in the generic sample function and document that it modifies the original array instead of creating a new one

I've never seen an array.length assignation (`array.length = numElements;`), it is not conventional, what is that for? If we create a new array with 10 elements (`slice`), and the original array reference is no longer accessible in the execution, the memory will be cleaned, so I won't worry about that.

Reply:
We make it predictable so that we can reproduce issues with customer integrations when they provide the logs to us. True, there is a bunch of inherent non-determinism in LLMs, but at least everything before that will be deterministic so we can debug the issues.

> I've never seen an array.length assignation (array.length = numElements;), it is not conventional
These dudes think it's quite acceptable: https://stackoverflow.com/a/31560542

> what is that for?
To change the length of an array without using extra memory.

Reply:
I could also change it to `splice` if that's more readable in your opinion. Though for me as someone not very familiar with JavaScript the meaning of `x.length = y` is immediately clear unlike `x.splice(...)`

Reply:
Mostly I don't want to return a slice because 

```
a = trimShuffleLogsSample(b)
```

will make the reader think that `trimShuffleLogsSample ` has no side effects and `b` is unchanged, which would be incorrect. While 

```
trimShuffleLogsSample(b)
```

makes it clear that `b` is changed in-place.

Reply:
So let's select between `splice()` and `length` assignment – do you think the first one is better?

Reply:
`splice` does return the deleted elements, which is something we don't need.

Reply:
Okay, all good. `length` is fine then 👍